proof_of_concept/policy/evaluation.py

```diff
@@ -222,24 +223,12 @@ def set_input_assets_permissions(
 
             This modifies the permissions argument.
             """
-            for inp_asset in job.inputs.values():
+            for inp_name, inp_asset in job.inputs.items():
                 if inp_asset not in permissions:
-                    permissions[inp_asset] = (
+                    permissions[inp_name] = (
                             self._policy_evaluator.permissions_for_asset(
                                 inp_asset))
 
-        def prop_workflow_inputs(
```

Comment:
Removed because it was a bit ugly, the permissions are indexed by workflow item (e.g. workflow input, step input, step output, etc.) except that I also put in the job input assets, then propagated those to the inputs. Saves a bit of computation if you use the same input asset for multiple workflow inputs, but also makes the code less clean. So now I just look up the permissions for each workflow input and put them in directly, so that everything is indexed by workflow item and it all makes sense. Slightly less efficient but at this point that's not important (and will probably never be).

Reply:
Sounds sane!